src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -152,19 +180,111 @@ public void accumulate(View found) {
                 LayoutRule originalValue = new LayoutRule(newVerb, currentRules[newVerb]);
                 mOriginalValues.put(found, originalValue);
             }
-            setLayout(found, newVerb, newAnchorId);
+
+            try {
+                setLayout(found, newVerb, newAnchorId);
+            } catch (LayoutUpdateException e) {
+                throw e;
+            }
         }
 
         private void setLayout(View target, int verb, int anchorId) {
```

Comment:
This guy is also called by cleanup. So, I think make it a separate function is more convenient.